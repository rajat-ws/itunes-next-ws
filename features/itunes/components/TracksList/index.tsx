import { Skeleton } from "antd";
import styled from "styled-components";
import { get } from "lodash";
import { If } from "@app/common";
import { TrackItem, TrackResponse } from "../../api/getTracks";
import TrackCard from "../TrackCard";

interface TrackListProps {
  tracksData: TrackResponse;
  loading: boolean;
}

const TracksListWrapper = styled.div`
  && {
    display: flex;
    gap: 2rem;
    flex-wrap: wrap;
    justify-content: space-evenly;
    margin: 0 auto;
  }
`;

const TracksList: React.FC<TrackListProps> = props => {
  const { tracksData, loading } = props;
  const results: TrackItem[] = get(tracksData, "results", []);

  return (
    <TracksListWrapper>
      <If condition={results.length !== 0}>
        <Skeleton loading={loading} active>
          {results.map((item: TrackItem, index: number) => {
            return <TrackCard key={index} data={item} />;
          })}
        </Skeleton>
      </If>
    </TracksListWrapper>
  );
};

export default TracksList;
