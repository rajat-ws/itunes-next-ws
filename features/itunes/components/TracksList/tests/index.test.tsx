import React from "react";
import TracksList from "../index";
import { render } from "@app/utils/testUtils";

describe("<TracksList />", () => {
  const trackListDataProps = {
    tracksData: {
      resultCount: 1,
      results: [
        {
          artistName: "Adele",
          collectionName: "collection",
          trackName: "Hello",
          trackId: 1544494392,
          artworkUrl100:
            "https://is5-ssl.mzstatic.com/image/thumb/Music116/v4/5a/47/6d/5a476ddd-4690-1297-1896-6a286a497a21/191404113974.png/100x100bb.jpg",
          previewUrl: "22233223",
        },
      ],
    },
    loading: true,
  };

  it("should render and match the snapshot", () => {
    const { baseElement } = render(<TracksList {...trackListDataProps} />);
    expect(baseElement).toMatchSnapshot();
  });
});
