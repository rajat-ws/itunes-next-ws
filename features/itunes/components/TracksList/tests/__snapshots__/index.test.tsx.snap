// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TracksList /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="TracksList__TracksListWrapper-sc-17ws8w0-0 GbUCs"
    >
      <div
        class="ant-skeleton ant-skeleton-active"
      >
        <div
          class="ant-skeleton-content"
        >
          <h3
            class="ant-skeleton-title"
            style="width: 38%;"
          />
          <ul
            class="ant-skeleton-paragraph"
          >
            <li />
            <li />
            <li
              style="width: 61%;"
            />
          </ul>
        </div>
      </div>
    </div>
  </div>
</body>
`;
